### PR TITLE
Abstract NessieClientBuilder + Move HTTP client to separate package

### PR DESCRIPTION
**Pull request by snazy**
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1816)

### CODE REVIEW

File: clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    return this;
+  }
+
+  /**
+   * Set the password for {@link AuthType#BASIC} authentication.
+   *
+   * @param password password
+   * @return {@code this}
+   */
+  public HttpClientBuilder withPassword(String password) {
```
Comment (dimas-b) [inline clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java:186]: Would it make sense to move auth parameters to the base `NessieClientBuilder` interface? I imagine we ought to support the same set of credential types across all clients (even though the credential transport may have to be different).

Comment (snazy) [reply clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java:189]: Right, those should not really differ semantically. Added that.

**Review by dimas-b — APPROVED** [review]

**Review by nastra — APPROVED** [review]
LGTM, the only thing that I noticed is that there's already a `HttpClient` in the `http` package, so that might cause confusion as having a `HttpClientBuilder` could imply that one creates a `HttpClient`. Not sure if we should name the builder `NessieHttpClientBuilder` or so

File: clients/client/src/main/java/org/projectnessie/client/NessieClientBuilder.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import java.net.URI;
+import java.util.function.Function;
+import org.projectnessie.client.NessieClient.AuthType;
+
+/**
+ * {@link NessieClient} builder interface.
+ *
+ * @param <IMPL> concrete nessie-client builder type
+ */
+public interface NessieClientBuilder<IMPL extends NessieClientBuilder<IMPL>> {
```
Comment (rymurr) [inline clients/client/src/main/java/org/projectnessie/client/NessieClientBuilder.java:27]: we discussed a protocol version and another parameter here also right?

Is this the only (future) way to get a Nessie client?

File: clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.http;
+
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_AUTH_TYPE;
```
Comment (rymurr) [inline clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java:18]: some of these constants are beyond the scope of just http client right? eg auth stuff, tracing

File: clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_OWNER;
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_PASSWORD;
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_REPOSITORY;
```
Comment (rymurr) [inline clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java:21]: can we just do a straight remove of all the repo/owner stuff. Nothing downstream is using it. I think it just adds confusion here.

**Review by rymurr — COMMENTED** [review]
agreed w/ @nastra re renaming and a few other comments

File: clients/client/src/main/java/org/projectnessie/client/NessieClientBuilder.java (hunk 2 above)
Comment (snazy) [reply clients/client/src/main/java/org/projectnessie/client/NessieClientBuilder.java:27]: Yes, this should become the only way to build+get a Nessie client. The current `NessieClient.Builder` shall go away.

Once this one and a follow-up to add the "API version" + "API invocation builder pattern" are in, we can nuke `NessieClient` + the current `TreeApi` + `ConfigApi` + `ContentsApi` interfaces. But it's a long(er) way to get there.

File: clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java (hunk 4 above)
Comment (snazy) [reply clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java:21]: Can do it in an immediate follow-up.

File: clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java (hunk 3 above)
Comment (snazy) [reply clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java:18]: yes

**Pull request merged**